backend/app/api/endpoints/share.py

```
@@ -0,0 +1,491 @@
+# =============================================================================
+
+
+@router.post(
+    "/{resource_type}/{resource_id}/link",
+    response_model=ShareLinkResponse,
+    summary="Create or update share link",
+)
+def create_share_link(
+    resource_type: str,
+    resource_id: int,
+    body: ShareLinkCreate = ShareLinkCreate(),
```

Ruff (0.14.14)

[warning] Do not perform function call `ShareLinkCreate` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)